pysatMadrigal/instruments/templates/madrigal_pandas.py

```diff
@@ -133,98 +124,58 @@ def init(self):
 
+    self.inst_code = self.kwargs['inst_code']
+    self.kindat = self.kwargs['kindat']
 
-def _general_download(date_array, tag='', sat_id='', data_path=None, user=None,
-                      password=None, inst_code=None, kindat=None):
-    """Downloads data from Madrigal.
-
-    Method will be partially set using functools.partial. Intended to
-    have the same call structure as normal instrument download routine.
-    Upon Instrument instantiation this routine will be set to
-    parameters specific to a Madrigal data set. It will then work like
-    a standard download call.
-
-    Parameters
-    ----------
-    date_array : array-like
-        list of datetimes to download data for. The sequence of dates need not
-        be contiguous.
-    tag : string ('')
-        Tag identifier used for particular dataset. This input is provided by
-        pysat.
-    sat_id : string  ('')
-        Satellite ID string identifier used for particular dataset. This input
-        is provided by pysat.
-    data_path : string (None)
-        Path to directory to download data to.
-    user : string (None)
-        User string input used for download. Provided by user and passed via
-        pysat. If an account
-        is required for dowloads this routine here must error if user not
-        supplied.
-    password : string (None)
-        Password for data download.
-    inst_code : int
-        Madrigal integer code used to identify platform
-    kindat : int
-        Madrigal integer code used to identify data set
-
-    Returns
-    --------
-    Void : (NoneType)
-        Downloads data to disk.
-
-    Notes
-    -----
-    The user's names should be provided in field user. Ruby Payne-Scott should
-    be entered as Ruby+Payne-Scott
-
-    The password field should be the user's email address. These parameters
-    are passed to Madrigal when downloading.
-
-    The affiliation field is set to pysat to enable tracking of pysat
-    downloads.
-
-    """
-    mad_meth.download(date_array, inst_code=inst_code, kindat=kindat,
-                      data_path=data_path, user=user, password=password)
+    return
 
 
 def clean(self):
     """Placeholder routine that would normally return cleaned data
 
-    Returns
-    --------
-    Void : (NoneType)
-        data in inst is modified in-place.
-
-    Notes
-    --------
+    Note
+    ----
     Supports 'clean', 'dusty', 'dirty' in the sense that it prints
     a message noting there is no cleaning.
     'None' is also supported as it signifies no cleaning.
 
     Routine is called by pysat, and not by the end user directly.
 
     """
+
     if self.clean_level in ['clean', 'dusty', 'dirty']:
         logger.warning('Generalized Madrigal data support has no cleaning.')
 
     return
+
+
+# ----------------------------------------------------------------------------
+# Instrument functions
+#
+# Use the default Madrigal and pysat methods
+
+# Set the list_remote_files routine
+# Need to fix this
```

Comment:
What does a user need to fix when creating a new instrument?

Reply:
The whole template needs to be fixed.  It's already an issue #1